docs/papers/architecture-whitepaper.md

```diff
@@ -0,0 +1,1709 @@
+                Rel(kmsService, kmsDb, "uses", "DB")
+            }
+
+        UpdateRelStyle(kmsService, tee, $textColor="green", $lineColor="blue", $offsetY="-30", $offsetX="10")
+        UpdateRelStyle(tee, hsm, $textColor="green", $lineColor="blue", $offsetX="-35", $offsetY="10")
+        UpdateRelStyle(kmsService, kmsDb, $textColor="green", $lineColor="blue", $offsetY="-30")
+        UpdateElementStyle(kmsDb, $fontColor="white", $bgColor="blue", $borderColor="blue")
+```
+
+###### Sub-capabilities
+- **Hardware Security Modules (HSMs)** - Dedicated tamper-resistant hardware devices designed specifically for storing and managing cryptographic keys. They offer the highest level of security due to physical isolation and protection against software attacks.
+- **Secure Enclaves** - Dedicated hardware or software components within a computing system that provide a secure execution environment for sensitive operations like key generation and storage.
+- **Key Management Systems** - Software solutions that centralize key management tasks like generation, storage, access control, and lifecycle management.
+- **Key Encryption** - Encryption of cryptographic keys using master keys or key-encryption keys (KEKs) before storing them in persistent storage. Key encryption ensures that even if the storage medium is compromised, encrypted keys remain protected and cannot be used without decryption.
+
+###### Considerations
+- **Compliance and Governance** - Compliance with regulatory requirements and industry standards governing cryptographic key management, such as PCI DSS, GDPR, or FIPS 140-2, should be ensured by implementing robust security controls and audit mechanisms.
+- **Hierarchical Deterministic (HD) Keys** - enable many public/private key pairs to be generated from a single seed, such as a 12- or 24-word mnemonic phrase.
+- **Shamir’s Secret Sharing (SSS)** - Consider whether a secret sharing scheme where a key is split into multiple shares, and any pre-defined number of shares are required to reconstruct the original key should be utilized.
+
+##### Credential Storage
+**Credential Storage** refers to the ability to store and manage credentials. Credentials come in many different standardized formats (e.g., SD-JWT, W3C Verifiable Credentials, MDOC), as well as different encoding schemes (e.g., JSON, JSON-LD, CBOR). Credential Storage makes use of the [Data Storage](#data-storage) capability and sub-capabilities, as credentials are a specialized form of data. Specific capabilities needed for credential storage include status management and schema and definition retrieval.
+
+![Credential Storage](./images/credential-storage.png "Credential Storage")
+
+###### Sub-capabilities
+- **Status Management** - Ability to expose and manage the revocation status and expiration dates of the credential.
+- **Schema and Definition Retrieval** - Ability to retrieve the credential schemas and definitions from the verifiable data registry or the linked data.
```

Comment:
It is mentioned that schemas and definitions come from the VDR but in the following part there are mentioned as being part of the trust registry.

Reply:
This may be a problem with mixing terminology (and may be a problem throughout the document). I will take a look at the entire document and update to reflect wording. Thanks for the catch.